src/handlers/execute.rs

```diff
@@ -256,3 +257,25 @@ fn refill_task(
 
     Ok(app.tag_response(Response::new().add_message(msg), "refill_task"))
 }
+
+fn purge(
+    deps: DepsMut,
+    _env: Env,
+    msg_info: MessageInfo,
+    app: CroncatApp,
+    task_tags: Vec<String>,
+) -> CroncatResult {
+    // In case module got unregistered or admin got changed they have no reason to purge now
+    if app
+        .admin
+        .assert_admin(deps.as_ref(), &msg_info.sender)
+        .is_err()
+    {
+        assert_module_installed(deps.as_ref(), &msg_info.sender, &app)?;
+    }
+
+    for tag in task_tags {
+        ACTIVE_TASKS.remove(deps.storage, (msg_info.sender.clone(), tag));
```

Comment:
@CyberHoward each module should be able to purge only his own tasks right?

Reply:
Yup, that would make sense!